src/worker.jl

```diff
@@ -129,6 +131,8 @@ function process_message(msg::Solve,
 
     update_ph_terms(sub.problem, msg.w_vals, msg.xhat_vals)
 
+    _execute_subproblem_callbacks(record, msg.niter)
```

Comment:
This gets called for every solve message that is received.  There is a solve message for each scenario.  `execute_subproblem_callbacks` seems to execute all callbacks on all subproblems.  This is why it gets called 12 times total.